### PR TITLE
Add test for global append dimension.

### PR DESCRIPTION
**Pull request by taohungyang**
# Description of the issue
This test makes sure ec2tagger tags metrics with the expected dimensions when users configure `append_dimensions` at the global metrics level.

# Description of changes
- Add a test for `append_dimensions` at global metrics level

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Test succeeded against my fork 
https://github.com/aws/private-amazon-cloudwatch-agent-staging/actions/runs/4046731013/jobs/6963800567

### CODE REVIEW

File: test/metric_dimension/global_append_dimensions_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			Value: dimension.UnknownDimensionValue(),
+		},
+		{
+			Key:   "InstanceId",
+			Value: dimension.UnknownDimensionValue(),
+		},
+		{
+			Key:   "InstanceType",
+			Value: dimension.UnknownDimensionValue(),
+		},
```
Comment (SaxyPandaBear) [inline test/metric_dimension/global_append_dimensions_test.go:65]: Does this work? We're using append dimensions in the config so these should have actual values. When we query for an unknown dimension here, it's going to return nothing, isn't it?

File: test/metric_dimension/global_append_dimensions_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+func (t *GlobalAppendDimensionsTestRunner) GetAgentConfigFileName() string {
+	return "global_append_dimension.json"
+}
+
+func (t *GlobalAppendDimensionsTestRunner) GetMeasuredMetrics() []string {
+	return []string{"cpu_time_active"}
+}
+
+func (t *GlobalAppendDimensionsTestRunner) validateNoAppendDimensionMetric(metricName string) status.TestResult {
```
Comment (SaxyPandaBear) [inline test/metric_dimension/global_append_dimensions_test.go:47]: nit - function name doesn't reflect what this is doing.

File: internal/awsservice/imds.go [hunk 3]
```diff
@@ -11,8 +11,20 @@ import (
...
+func GetImageId() string {
+	return GetImdsMetadata().ImageID
+}
+
+func GetInstanceType() string {
+	return GetImdsMetadata().InstanceType
+}
+
+// TODO: Refactor Structure and Interface for more easier follow that shares the same session
+func GetImdsMetadata() *imds.GetInstanceIdentityDocumentOutput {
```
Comment (SaxyPandaBear) [inline internal/awsservice/imds.go:27]: We should cache the results of the metadata call so we don't ping IMDS a lot.

File: test/metric_dimension/global_append_dimensions_test.go (hunk 1 above)
Comment (taohungyang) [reply test/metric_dimension/global_append_dimensions_test.go:65]: I checked the providers. As you can see here:
https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/test/metric/dimension/instanceid_provider.go#L59-L62
It's only when the value is unknown that the provider (instanceID/imageId/InstanceType) will retrieve the actual value from imds and use it to query the metrics

Comment (SaxyPandaBear) [reply test/metric_dimension/global_append_dimensions_test.go:65]: Ohh I guess I misunderstood how it works

File: internal/awsservice/imds.go (hunk 3 above)
Comment (taohungyang) [reply internal/awsservice/imds.go:29]: Added a variable to cached the results. Thanks

Comment (sethAmazon) [reply internal/awsservice/imds.go:29]: Do we plan on calling this multiple times within the same test execution, that is needs to be cached?

**Review by sethAmazon — DISMISSED** [review]

File: internal/awsservice/imds.go [hunk 4]
```diff
@@ -11,8 +11,25 @@ import (
...
+	if indentityDoc != nil {
```
Comment (jefchien) [inline internal/awsservice/imds.go:30]: nit: misspelled
```suggestion
	if identityDoc != nil {
```

Comment (taohungyang) [reply internal/awsservice/imds.go]: Good eye...fixed it.

**Review by ChenaLee — APPROVED** [review]
This is awesome. Thanks!

**Pull request merged**